pkg/endpoint/endpoint.go

```diff
@@ -2752,9 +2752,8 @@ func (e *Endpoint) garbageCollectConntrack(filter *ctmap.GCFilter) {
 			log.WithError(err).WithField(logfields.Path, filepath).Warn("Unable to open map")
 			continue
 		}
-		defer m.Close()
-
 		ctmap.GC(m, filter)
+		m.Close()
```

Comment:
This closes the FD earlier but how does this fix a memory leak? The range over maps is finite and the deferred `Close()` will be invoked.

Reply:
@tgraf good point, the bug is more likely to be #6024